cpp/src/parquet/bloom_filter.cc

```diff
@@ -146,7 +208,7 @@ bool BlockSplitBloomFilter::FindHash(uint64_t hash) const {
 
 void BlockSplitBloomFilter::InsertHash(uint64_t hash) {
   const uint32_t bucket_index =
-      static_cast<uint32_t>(hash >> 32) & (num_bytes_ / kBytesPerFilterBlock - 1);
+      static_cast<uint32_t>(((hash >> 32) * (num_bytes_ / kBytesPerFilterBlock)) >> 32);
   uint32_t key = static_cast<uint32_t>(hash);
```

Comment:
`const` here too?

Reply:
Seems not, `Insert` may modify bitset...